### PR TITLE
Support `x-tracking-request-id` header to have separate parallel data requests.

### PR DESCRIPTION
**Pull request by zeetabit**
- Introduced `X-Tracking-Request-Id` header.
- Introduced concurrent request handling support by dividing data by introduced header.
- Introduced unit tests with new header.

### CODE REVIEW

**Review by DanielHansmannAldi — CHANGES_REQUESTED** [review]
I’m a little bit confused about the Methode and variable naming.

If I get the Idea right it is to enable parallel requests of the same kind and therefore add a requestID. (What I totally like by the wax 👍)

so can we please find a consistent naming for this.

Method is confusing because of HTTP Method

**Comment by zeetabit** [comment]
@DanielHansmannAldi done, please review one more time.

File: src/phpMockServer.php [hunk 1]
```diff
@@ -267,7 +291,14 @@ private function readMockedRequest()
             $this->response->setStatusCode(404);
             return;
         }
+        $requestId = $this->getRequestId();
```
Comment (DanielHansmannAldi) [inline src/phpMockServer.php]: You repeat yourself with line 210

Comment (DanielHansmannAldi) [inline src/phpMockServer.php]: You repeat yourself with line 210

File: src/rules/preselectionRuleImplementation.php [hunk 2]
```diff
@@ -24,10 +32,16 @@ public static function check(\Symfony\Component\HttpFoundation\Request &$request
         return true;
     }
 
-    protected static function readPreselection($methode, $path)
+    protected static function readPreselection($methode, $xMockRequestId, $path)
```
Comment (DanielHansmannAldi) [inline src/rules/preselectionRuleImplementation.php]: Why Donnie make this a breaking change?
Why not move to the end and make it optional?

File: src/phpMockServer.php (hunk 1 above)
Comment (zeetabit) [reply src/phpMockServer.php]: removed duplication

**Comment by zeetabit** [comment]
@DanielHansmannAldi done, please review one more time.

**Review by DanielHansmannAldi — APPROVED** [review]

**Pull request merged**